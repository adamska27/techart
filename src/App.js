import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import store from './store';
import { theme } from './styles/theme';

import HeaderContainer from './containers/Header';

import Collection from './pages/Collection';
import Game from './pages/Game';
import Home from './pages/Home';
import LogIn from './pages/LogIn';
import PagesNotFound from './pages/PagesNotFound';
import Profile from './pages/Profile';
import RatingsReviewForm from './pages/RatingsReviewForm';
import Review from './pages/Review';
import ReviewsAll from './pages/ReviewsAll';
import SignUp from './pages/SignUp';

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router>
            <React.Fragment>
              <HeaderContainer />
              <Switch>
                <Route exact path="/" component={Home} />
                <Route path="/game/:gameId" component={Game} />
                <Route path="/:gameId/rating" component={RatingsReviewForm} />
                <Route path="/signup" component={SignUp} />
                <Route path="/login" component={LogIn} />
                <Route path="/user/:id" component={Profile} />
                <Route path="/collection" component={Collection} />
                <Route exact path="/review/:id" component={Review} />
                <Route
                  path="/review/product/:productId"
                  component={ReviewsAll}
                />
                <Route component={PagesNotFound} />
              </Switch>
            </React.Fragment>
          </Router>
        </ThemeProvider>
      </Provider>
    );
  }
}

export default App;
