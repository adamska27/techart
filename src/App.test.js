import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('match the snapshot', () => {
  const component = renderer.create(<App />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
