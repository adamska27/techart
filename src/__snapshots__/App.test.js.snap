// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match the snapshot 1`] = `
Array [
  <header
    className="sc-bwzfXH fcWfTy"
>
    <nav
        className="sc-htpNat ljNRKX"
    >
        <a
            href="/"
            onClick={[Function]}
        >
            <div
                className="sc-bxivhb oNkbj"
            >
                Logo
            </div>
        </a>
        <ul
            className="sc-ifAKCX eRcwmR"
        >
            <a
                href="/"
                onClick={[Function]}
            >
                <li>
                    Accueil
                </li>
            </a>
            <a
                href="/games"
                onClick={[Function]}
            >
                <li>
                    Jeux
                </li>
            </a>
            <a
                href="/members"
                onClick={[Function]}
            >
                <li>
                    membres
                </li>
            </a>
            <a
                href="/mygames"
                onClick={[Function]}
            >
                <li>
                    mes jeux
                </li>
            </a>
        </ul>
        <ul
            className="sc-bdVaJa eEQXFw"
        >
            <a
                href="/signup"
                onClick={[Function]}
            >
                <li>
                    Inscription
                </li>
            </a>
            <a
                href="/signin"
                onClick={[Function]}
            >
                <li>
                    Connexion
                </li>
            </a>
        </ul>
    </nav>
</header>,
  <p>
    Loading
</p>,
]
`;
